executors/multiversx/refundExecutor.go

```diff
@@ -0,0 +1,165 @@
+			transactionExecutor:    args.TransactionExecutor,
+			codec:                  args.Codec,
+			filter:                 args.Filter,
+			log:                    args.Log,
+		},
+		gasToExecute: args.GasToExecute,
+	}
+
+	err := executor.checkBaseComponents()
+	if err != nil {
+		return nil, err
+	}
+
+	return executor, nil
+}
+
+// Execute will execute one step: get all pending operations, call the filter and send execution transactions
+func (executor *refundExecutor) Execute(ctx context.Context) error {
+	return executor.executeOnAllScProxyAddress(ctx, executor.executeRefundForScProxyAddress)
+}
+
+func (executor *refundExecutor) executeRefundForScProxyAddress(ctx context.Context, scProxyAddress string) error {
+	executor.log.Info("Executing refunds for the SC proxy address", "address", scProxyAddress)
+
+	pendingOperations, err := executor.getPendingRefunds(ctx, scProxyAddress)
+	if err != nil {
+		return err
+	}
+
+	filteredPendingOperations := executor.filterOperations("refundExecutor", pendingOperations)
+
+	return executor.executeRefunds(ctx, filteredPendingOperations, scProxyAddress)
+}
+
+func (executor *refundExecutor) getPendingRefunds(ctx context.Context, scProxyAddress string) (map[uint64]bridgeCore.ProxySCCompleteCallData, error) {
+	request := &data.VmValueRequest{
+		Address:  scProxyAddress,
+		FuncName: getRefundTransactionsFunction,
+	}
+
+	response, err := executor.proxy.ExecuteVMQuery(ctx, request)
+	if err != nil {
+		executor.log.Error("got error on VMQuery", "FuncName", request.FuncName,
+			"Args", request.Args, "SC address", request.Address, "Caller", request.CallerAddr, "error", err)
+		return nil, err
+	}
+	if response.Data.ReturnCode != okCodeAfterExecution {
+		return nil, errors.NewQueryResponseError(
+			response.Data.ReturnCode,
+			response.Data.ReturnMessage,
+			request.FuncName,
+			request.Address,
+			request.Args...,
+		)
+	}
+
+	return executor.parseResponse(response)
+}
+
+func (executor *refundExecutor) parseResponse(response *data.VmValuesResponseData) (map[uint64]bridgeCore.ProxySCCompleteCallData, error) {
+	numResponseLines := len(response.Data.ReturnData)
+	if numResponseLines%2 != 0 {
+		return nil, fmt.Errorf("%w: expected an even number, got %d", errInvalidNumberOfResponseLines, numResponseLines)
+	}
+
+	result := make(map[uint64]bridgeCore.ProxySCCompleteCallData, numResponseLines/2)
+
+	for i := 0; i < numResponseLines; i += 2 {
+		pendingOperationID := big.NewInt(0).SetBytes(response.Data.ReturnData[i])
+		callData, err := executor.codec.DecodeProxySCCompleteCallData(response.Data.ReturnData[i+1])
+		if err != nil {
+			return nil, fmt.Errorf("%w for ReturnData at index %d", err, i+1)
+		}
+
+		result[pendingOperationID.Uint64()] = callData
+	}
+
+	return result, nil
+}
+
+func (executor *refundExecutor) executeRefunds(ctx context.Context, refundIDs map[uint64]bridgeCore.ProxySCCompleteCallData, scProxyAddress string) error {
+	networkConfig, err := executor.proxy.GetNetworkConfig(ctx)
```

Comment:
why not cache the config?

Reply:
I guess the SDK already caches the config, as seen here https://github.com/multiversx/mx-sdk-go/blob/888942f27039e743ad20e1169637a3da99abbb99/blockchain/baseProxy.go#L75